pylint/checkers/similar.py

```diff
@@ -26,32 +26,315 @@
+        return (
+            f"<LinesChunk object for file {self._fileid} ({self._index}, {self._hash})>"
+        )
+
+    def __str__(self) -> str:
+        return (
+            f"LinesChunk object for file {self._fileid}, starting at line {self._index} \n"
+            f"Hash is {self._hash}"
+        )
+
+
+class SuccessiveLinesLimits:
+    """
+    A class to handle the numbering of begin and end of successive lines.
+
+    :note: Only the end line number can be updated.
+    """
+
+    def __init__(self, start: LineNumber, end: LineNumber):
+        self._start: LineNumber = start
+        self._end: LineNumber = end
+
+    @property
+    def start(self) -> LineNumber:
+        return self._start
+
+    @property
+    def end(self) -> LineNumber:
+        return self._end
+
+    @end.setter
+    def end(self, value: LineNumber):
+        self._end = value
+
+    def __repr__(self) -> str:
+        return f"<SuccessiveLinesLimits <{self._start};{self._end}>>"
+
+
+class LineSetStartCouple(NamedTuple):
+    """
+    Indices in both linesets that mark the beginning of successive lines
+    """
+
+    fst_lineset_index: Index
+    snd_lineset_index: Index
+
+    def __repr__(self) -> str:
+        return (
+            f"<LineSetStartCouple <{self.fst_lineset_index};{self.snd_lineset_index}>>"
+        )
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, LineSetStartCouple):
+            raise NotImplementedError
+        return (
+            self.fst_lineset_index == other.fst_lineset_index
+            and self.snd_lineset_index == other.snd_lineset_index
+        )
+
+    def __hash__(self) -> int:
+        return hash(self.fst_lineset_index) + hash(self.snd_lineset_index)
+
+    def increment(self, value: Index) -> "LineSetStartCouple":
+        return LineSetStartCouple(
+            Index(self.fst_lineset_index + value),
+            Index(self.snd_lineset_index + value),
+        )
+
+
+def hash_lineset(
+    lineset: "LineSet", min_common_lines: int = DEFAULT_MIN_SIMILARITY_LINE
+) -> Tuple[HashToIndex_T, IndexToLines_T]:
+    """
+    Return two dicts. The first associates the hash of successive stripped lines of a lineset
+    to the indices of the starting lines.
+    The second dict, associates the index of the starting line in the lineset's stripped lines to the
+    couple [start, end] lines number in the corresponding file.
+
+    :param lineset: lineset object (i.e the lines in a file)
+    :param min_common_lines: number of successive lines that are used to compute the hash
+    :return: a dict linking hashes to corresponding start index and a dict that links this
+             index to the start and end lines in the file
+    """
+    hash2index = defaultdict(list)
+    index2lines = dict()
+    # Comments, docstring and other specific patterns maybe excluded -> call to stripped_lines
+    # to get only what is desired
+    lines = list(x.text for x in lineset.stripped_lines)
```

Comment:
```suggestion
    lines = tuple(x.text for x in lineset.stripped_lines)
```
Does a tuple work here?

Reply:
Yes! Good idea!